monai/visualize/img2tensorboard.py

```diff
@@ -67,31 +77,40 @@ def make_animated_gif_summary(
     axis_order = [0] + animation_axes + image_axes
 
     slicing = []
-    for i in range(len(tensor.shape)):
+    for i in range(len(image.shape)):
         if i in axis_order:
             slicing.append(slice(None))
         else:
             other_ind = other_indices.get(i, 0)
             slicing.append(slice(other_ind, other_ind + 1))
-    tensor = tensor[tuple(slicing)]
+    image: Union[Tensor, np.array] = image[tuple(slicing)]
 
-    for it_i in range(min(max_out, list(tensor.shape)[0])):
-        inp = [tag + suffix.format(it_i), tensor[it_i, :, :, :]]
-        summary_op = _image3_animated_gif(inp, scale_factor)
+    for it_i in range(min(max_out, list(image.shape)[0])):
+        one_channel_img: Union[Tensor, np.array] = image[it_i, :, :, :].squeeze(dim=0) if torch.is_tensor(
```

Comment:
the `squeeze` here makes tensor/nparray processed differently, not sure if this is intended. would be great to have a unit test if possible.

Reply:
@wyli I don't know how to make a unit test that is meaning full for these tests.  I did run these functions to create a tensorboard visualization that worked as expected.  I don't think it will be a trivial task to create a tensorboard visualization unit test.

Refactoring to make the array processing separate from the tensorboard Summary object generation would be possible, but is a lot more invasive refactoring.